projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -0,0 +1,1876 @@
+			ret |= AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+						   &cn0411_dev->ad7124_dev.regs[AD7124_Status]);
+			status_reg = cn0411_dev->ad7124_dev.regs[AD7124_Status].value
+				     & ADC_CH_RDY_MSK;
+		}
+		while(status_reg != ADC_CH0) {
+			ret |= AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+						   &cn0411_dev->ad7124_dev.regs[AD7124_Status]);
+			status_reg = cn0411_dev->ad7124_dev.regs[AD7124_Status].value
+				     & ADC_CH_RDY_MSK;
+		}
+		ret |= AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+					   &cn0411_dev->ad7124_dev.regs[AD7124_Data]);
+		ch0_data = cn0411_dev->ad7124_dev.regs[AD7124_Data].value;
+		break;
+	case ADC_SINGLE_CONV:
+		ret = CN0411_ADC_ch0 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_operation_mode(cn0411_dev, SINGLE_CONV);
+		if (AD7124_WaitForConvReady(&cn0411_dev->ad7124_dev, ADC_TIMEOUT)
+		    == -3) {
+			printf("TIMEOUT\n");
+			return -1;
+		}
+		ret |= AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+					   &cn0411_dev->ad7124_dev.regs[AD7124_Data]);
+		ch0_data = cn0411_dev->ad7124_dev.regs[AD7124_Data].value;
+		ret |= CN0411_ADC_ch0 (cn0411_dev, ADC_CH_DISABLE);
+		break;
+	default:
+		ret = CN0411_FAILURE;
+		break;
+	}
+	float resistance = (float)(ch0_data) * RTD_REF_RES / 0xFFFFFF;
+	if(resistance < cn0411_dev->rtd_res)
+		cn0411_dev->temp = -242.02 + 2.228 * resistance + (2.5859 * pow(10, -3))
+				   * pow(resistance, 2) - (48260.0 * pow(10, -6))
+				   * pow(resistance, 3) - (2.8183 * pow(10, -3))
+				   * pow(resistance, 4) + (1.5243 * pow(10, -10))
+				   * pow(resistance, 5);
+	else
+		cn0411_dev->temp = (-A + sqrt((pow(A, 2) - 4 * B * (1 - resistance
+					       / cn0411_dev->rtd_res)))) / (2 * B);
+
+	return ret;
+}
+
+/**
+ * CN0411 read peak-to-peak voltage
+ *
+ * Reads channel 1 and 2 of the ADC
+ * Based on the conversion mode used:
+ * 	- continuous: since the function is called by the user and the sequence
+ * 	  of channels conversions cannot be controlled individually, the function
+ * 	  assures that the conversion data is read properly, gathering the data only
+ * 	  after the transition is made on the desired ADC channel. Therefore, if the
+ * 	  currently converted ADC channel is channel 1, the	program waits until next
+ * 	  channel conversion is made and reads the data register when transition to
+ * 	  channel 1 is detected. Then data register is read again when transition to
+ * 	  channel 2 is detected.
+ * 	- single: read channel 1 and 2 individually
+ *
+ * @param cn0411_dev - The device structure.
+
+ * @return peak-to-peak voltage value
+*/
+int32_t CN0411_read_vpp(cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+	uint32_t ch1_data = 0, ch2_data = 0, status_reg;
+
+	switch (cn0411_dev->conv_type) {
+	case ADC_CONTINUOUS_CONV:
+		ret = AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+					  &cn0411_dev->ad7124_dev.regs[AD7124_Status]);
+		status_reg = cn0411_dev->ad7124_dev.regs[AD7124_Status].value
+			     & ADC_CH_RDY_MSK;
+		while(status_reg == ADC_CH1) {
```

Comment:
This construction seems to be repeated a lot in the code. You could create a function with the channel as a parameter.